frontend/lib/src/components/core/Block/utils.test.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ElementNode } from "@streamlit/lib/src/AppNode"
+import { ScriptRunState } from "@streamlit/lib/src/ScriptRunState"
+
+import { isElementStale } from "./utils"
+
+describe("isElementStale", () => {
+  // @ts-expect-error
+  const node = new ElementNode(null, null, "myScriptRunId", "myFragmentId")
+
+  it("Returns true if scriptRunState is RERUN_REQUESTED", () => {
+    expect(
+      isElementStale(node, ScriptRunState.RERUN_REQUESTED, "someScriptRunId")
+    ).toBe(true)
+  })
+
+  it("If running and currentFragmentId is set, compares with node's fragmentId", () => {
+    expect(
+      isElementStale(
+        node,
+        ScriptRunState.RUNNING,
+        "myScriptRunId",
+        "myFragmentId"
+      )
+    ).toBe(true)
+
+    expect(
+      isElementStale(
+        node,
+        ScriptRunState.RUNNING,
+        "myScriptRunId",
+        "someOtherFragmentId"
+      )
+    ).toBe(false)
+  })
+
+  it("If running and currentFragmentId is not set, compares with node's scriptRunId", () => {
+    expect(
+      isElementStale(node, ScriptRunState.RUNNING, "someOtherScriptRunId")
```

Comment:
is this `stale` because no script run id `someOtherScriptRunId` exists? I feel it might helpful to have a small comment explaining why its true for this id and false for the other for those who are not too deep in the library (yet) 🙂

Reply:
👍 added an explanatory comment